src/main/java/org/prebid/server/bidder/datablocks/DatablocksBidder.java

```diff
@@ -0,0 +1,150 @@
+package org.prebid.server.bidder.datablocks;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.Json;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.datablocks.ExtImpDatablocks;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class DatablocksBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpDatablocks>> DATABLOCKS_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpDatablocks>>() {
+            };
+
+    private final String endpointTemplate;
+
+    public DatablocksBidder(String endpointTemplate) {
+        this.endpointTemplate = HttpUtil.validateUrl(Objects.requireNonNull(endpointTemplate));
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        final Map<ExtImpDatablocks, List<Imp>> extToImps = new HashMap<>();
+        for (Imp imp : bidRequest.getImp()) {
+            try {
+                final ExtImpDatablocks extImpDatablocks = parseAndValidateImpExt(imp.getExt());
+                extToImps.putIfAbsent(extImpDatablocks, new ArrayList<>());
+                extToImps.get(extImpDatablocks).add(imp);
```

Comment:
Probably you can inline to `.computeIfAbsent(extImpDatablocks, key -> new ArrayList<>()).add(imp)`